awscli/customizations/s3/utils.py

```diff
@@ -760,6 +760,33 @@ def on_queued(self, future, **kwargs):
         future.meta.provide_transfer_size(self.size)
 
 
+class OnDoneFilteredSubscriber(BaseSubscriber):
```

Comment:
Is there a reason for this not to be in s3transfer?

It's a little weird to me to add new public methods to a base class that look like they come from s3transfer but actually don't.

It might even make sense to just move this directly into `BaseSubscriber`.